Source/Core/DolphinQt/SkylanderPortal/SkylanderModifyDialog.cpp

```diff
@@ -0,0 +1,337 @@
+                    nickname.length());
+
+      m_figure->SetData(&m_figure_data);
+      this->accept();
+    }
+  });
+}
+
+void SkylanderModifyDialog::PopulateTrophyOptions()
+{
+  static constexpr u16 KAOS_TROPHY_ID = 3503;
+  static constexpr u16 SEA_TROPHY_ID = 3502;
+
+  if (m_figure_data.figure_id == KAOS_TROPHY_ID)
+  {
+    m_should_show = false;
+    QMessageBox::warning(this, tr("Can't edit villains for this trophy!"),
+                         tr("Kaos is the only villain for this trophy and is always unlocked. No "
+                            "need to edit anything!"),
+                         QMessageBox::Ok);
+    return;
+  }
+
+  auto* hbox_villain_1 = new QHBoxLayout();
+  auto* label_villain_1 = new QLabel(tr("Captured villain %1:").arg(1));
+  auto* edit_villain_1 = new QCheckBox();
+  edit_villain_1->setChecked(m_figure_data.trophy_data.unlocked_villains & 0b1);
+  hbox_villain_1->addWidget(label_villain_1);
+  hbox_villain_1->addWidget(edit_villain_1);
+  m_layout->addLayout(hbox_villain_1);
+
+  u8 shift_count = 1;
+  if (m_figure_data.figure_id != SEA_TROPHY_ID)
+    shift_count++;
+
+  auto* hbox_villain_2 = new QHBoxLayout();
+  auto* label_villain_2 = new QLabel(tr("Captured villain %1:").arg(2));
+  auto* edit_villain_2 = new QCheckBox();
+  edit_villain_2->setChecked(m_figure_data.trophy_data.unlocked_villains & (0b1 << shift_count));
+  hbox_villain_2->addWidget(label_villain_2);
+  hbox_villain_2->addWidget(edit_villain_2);
+  m_layout->addLayout(hbox_villain_2);
+
+  shift_count++;
+
+  auto* hbox_villain_3 = new QHBoxLayout();
+  auto* label_villain_3 = new QLabel(tr("Captured villain %1:").arg(3));
+  auto* edit_villain_3 = new QCheckBox();
+  edit_villain_3->setChecked(m_figure_data.trophy_data.unlocked_villains & (0b1 << shift_count));
+  hbox_villain_3->addWidget(label_villain_3);
+  hbox_villain_3->addWidget(edit_villain_3);
+  m_layout->addLayout(hbox_villain_3);
+
+  shift_count = 4;
+
+  auto* hbox_villain_4 = new QHBoxLayout();
+  auto* label_villain_4 = new QLabel(tr("Captured villain %1:").arg(4));
+  auto* edit_villain_4 = new QCheckBox();
+  edit_villain_4->setChecked(m_figure_data.trophy_data.unlocked_villains & (0b1 << shift_count));
+  hbox_villain_4->addWidget(label_villain_4);
+  hbox_villain_4->addWidget(edit_villain_4);
+  m_layout->addLayout(hbox_villain_4);
+
+  connect(m_buttons, &QDialogButtonBox::accepted, this, [=, this]() {
+    m_figure_data.trophy_data.unlocked_villains = 0x0;
+    m_figure_data.trophy_data.unlocked_villains |= (edit_villain_1->isChecked() ? 0b1 : 0b0);
+    m_figure_data.trophy_data.unlocked_villains |=
+        (edit_villain_2->isChecked() ? 0b1 << ((m_figure_data.figure_id == 3502) ? 1 : 2) : 0b0);
+    m_figure_data.trophy_data.unlocked_villains |=
+        (edit_villain_3->isChecked() ? 0b1 << ((m_figure_data.figure_id == 3502) ? 2 : 3) : 0b0);
+    m_figure_data.trophy_data.unlocked_villains |= (edit_villain_4->isChecked() ? (0b1 << 4) : 0b0);
```

Comment:
Continuing from above:

```suggestion
unlocked_villains = 0x0;
for (size_t i = 0; i < MAX_VILLAINS; ++i)
  unlocked_villains |= edit_villains[i]->isChecked ? (0b1 << shift_distances[i]) : 0b0;
```